src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -445,6 +450,22 @@ export function useAnimatedStyle<T extends AnimatedStyle>(
 
   dependencies.push(sharableViewDescriptors);
 
+  const areListsEqual = (a1: DependencyList, a2: DependencyList) => {
+    for (let i = 0; i < (a1?.length || 0); ++i) {
+      if (a1?.[i] !== a2?.[i]) return false;
+    }
+    return true;
+  };
+
+  // dependencies can change in length and React doesn't like that
+  // so in useEffect instead of passing just a list of dependencies we pass the list of dependencies itself as a dependency
```

Comment:
This doesn't sound like a correct approach. Passing a list likely means there will never be referential equality (each list is perhaps compiled during render), so this will result in this hook being run at every render which misses the point of passing any dependencies in the first place

Reply:
A few lines below there is a check:
```js
if (
    dependencies.length !== depsRef.current[0]?.length ||
    !areListsEqual(dependencies, depsRef.current)
  ) {
    depsRef.current = [dependencies];
  }
```
So if the length of dependencies changes or the dependencies itself change then we create a new list and assign it to `depsRef.current`. Then we pass `depsRef.current` as a hook dependency. If nothing changes in dependencies, we don't touch `depsRef.current` so it's the same list and hook is not run.